src/main/java/org/openRealmOfStars/starMap/Route.java

```diff
@@ -180,10 +113,9 @@ public boolean isFixing() {
    */
   public void makeNextMove() {
     for (int i = 0; i < ftlSpeed; i++) {
-      if (distance > 0) {
-        startX = startX + mx;
-        startY = startY + my;
-        distance--;
+      if (getDistance() > 0) {
+        startX = startX + getMx();
```

Comment:
The test cases that you wrote went green and now they are green still. The third test was unneeded, because that didn't add more logic/more code coverage, so I deleted it. The two others are still there I just a bit rewrite them to avoid loops in tests. And added more asserts. Could you make a test that is green with the previous solution and red with this? Or could you explain some repro steps?